Userland/Utilities/echo.cpp

```diff
@@ -9,6 +9,58 @@
 #include <stdio.h>
 #include <unistd.h>
 
+static char backslash_escaped_char(char c)
+{
+    switch (c) {
+    case '\\':
+        return c;
+    // `\"` produces `"` with printf(1), but `\"` with echo(1)
+    case 'a':
+        return '\a';
+    case 'b':
+        return '\b';
+    case 'e':
+        return '\e';
+    case 'f':
+        return '\f';
+    case 'n':
+        return '\n';
+    case 'r':
+        return '\r';
+    case 't':
+        return '\t';
+    case 'v':
+        return '\v';
+    default:
+        return c;
+    }
+}
+
+static String interpret_backslash_escapes(String s)
+{
+    StringBuilder builder;
+
+    for (size_t i = 0; i < s.length();) {
+        if (char c = s[i++]; c != '\\') {
+            builder.append(c);
+            continue;
+        }
+        if (i == s.length()) {
+            // Last character of string is '\' -- output it verbatim.
+            builder.append('\\');
+        }
+
+        char c = s[i++];
+        if (c == 'c') // `\c` suppresses further output.
+            break;
+        // FIXME: \0ooo, \xHH, \uHHHH, \UHHHHHHHH should produce characters if followed by
+        // enough digits.
+        builder.append(backslash_escaped_char(c));
```

Comment:
Could also use GenericLexer here to avoid implementing this, though it doesn't look particularly complicated, so this seems alright as well

Reply:
Keeping as-is works for me ;)